plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java

```diff
@@ -155,6 +155,55 @@ public void testReadWholePartition()
+        assertUpdate("INSERT INTO test_history_system_table VALUES (1, 2)", 1);
+        assertUpdate("INSERT INTO test_history_system_table VALUES (2, 3)", 1);
+        assertUpdate("INSERT INTO test_history_system_table VALUES (3, 4)", 1);
+        assertUpdate("INSERT INTO test_history_system_table VALUES (4, 5)", 1);
+
+        assertFileSystemAccesses("SELECT * FROM \"test_history_system_table$history\"",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000000.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000001.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000003.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000004.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000005.json", INPUT_FILE_NEW_STREAM))
+                        .build());
+
+        assertFileSystemAccesses("SELECT * FROM \"test_history_system_table$history\" WHERE version = 3",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000003.json", INPUT_FILE_NEW_STREAM))
+                        .build());
+
+        assertFileSystemAccesses("SELECT * FROM \"test_history_system_table$history\" WHERE version > 3",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000004.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000005.json", INPUT_FILE_NEW_STREAM))
+                        .build());
+
+        assertFileSystemAccesses("SELECT * FROM \"test_history_system_table$history\" WHERE version >= 3 OR version = 1",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000001.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000003.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000004.json", INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000005.json", INPUT_FILE_NEW_STREAM))
+                        .build());
+
+        assertFileSystemAccesses("SELECT * FROM \"test_history_system_table$history\" WHERE version >= 1 AND version < 3",
```

Comment:
I commented all the  ..$history  `assertFileSystemAccesses` checks and got the following error:

```
java.lang.AssertionError: Expected: 
		24 fewer occurrences of FileOperation[fileType=LAST_CHECKPOINT, fileId=_last_checkpoint, operationType=INPUT_FILE_NEW_STREAM],
		24 fewer occurrences of FileOperation[fileType=TRANSACTION_LOG_JSON, fileId=00000000000000000005.json, operationType=INPUT_FILE_NEW_STREAM]
```

Reply:
See also on CI that `trino-delta-lake` is actually failing.